mmit/core/piecewise_function.cpp

```diff
@@ -165,7 +165,7 @@ int PiecewiseFunction::insert_point(double y, bool is_upper_bound) {
     }
 
     // Breakpoint info
-    float b = y - s * this->margin;
```

Comment:
@tdhock I'm pretty sure this is why we had really poor precision on the breakpoint positions! The breakpoint positions were floats and were getting converted to double each time a comparison was made, since all the other real-valued variables have type double.